scripts/events_module/patrol/patrol_outcome.py

```diff
@@ -677,56 +676,58 @@ def _handle_herbs(self, patrol: "Patrol") -> str:
         if not self.herbs or game.clan.game_mode == "classic":
             return ""
 
+        list_of_herb_strs = []
+
         large_bonus = False
         if "many_herbs" in self.herbs:
             large_bonus = True
 
-        # Determine which herbs get picked
-        specific_herbs = [x for x in self.herbs if x in HERBS]
-        if "random_herbs" in self.herbs:
-            specific_herbs += random.sample(
-                HERBS, k=choices([1, 2, 3], [6, 5, 1], k=1)[0]
-            )
-
-        # Remove duplicates
-        specific_herbs = list(set(specific_herbs))
-
-        if not specific_herbs:
-            print(f"{self.herbs} - gave no herbs to give")
-            return ""
-
-        patrol_size_modifier = int(len(patrol.patrol_cats) * 0.5)
-        herb_names = []
-        for _herb in specific_herbs:
-            if large_bonus:
-                amount_gotten = 6
-            else:
-                amount_gotten = choices([2, 4, 6], [2, 3, 1], k=1)[0]
+        if len(patrol.patrol_cats) > 2:
+            patrol_size_modifier = int(len(patrol.patrol_cats) * 0.5)
+        else:
+            patrol_size_modifier = 1
 
-            amount_gotten = int(amount_gotten * patrol_size_modifier)
-            amount_gotten = max(1, amount_gotten)
+        full_amount_count = 0
 
-            if _herb in game.clan.herbs:
-                game.clan.herbs[_herb] += amount_gotten
-            else:
-                game.clan.herbs[_herb] = amount_gotten
+        if "random_herbs" in self.herbs:
+            list_of_herb_strs, found_herbs = game.clan.herb_supply.get_found_herbs(
+                med_cat=patrol.patrol_leader,
+                general_amount_bonus=large_bonus,
+                specific_quantity_bonus=patrol_size_modifier
+            )
+        else:
+            found_herbs = {}
+            for herb in [x for x in self.herbs if x not in ["many_herbs", "random_herbs"]]:
+                amount = choices([2, 3, 4], weights=[2, 1, 1], k=1)[0]
+                amount *= patrol_size_modifier
+                if large_bonus:
+                    amount *= 2
+
+                found_herbs[herb] = amount
+
+            for herb, count in found_herbs.items():
+                game.clan.herb_supply.add_herb(herb, count)
+                full_amount_count += count
+                if count > 1:
+                    list_of_herb_strs.append(f"{count} {game.clan.herb_supply.herb[herb].plural_display}")
```

Comment:
This bothers me again for localization reasons, but I say we leave it and if it turns out to be an issue for localizers we can revisit